src/toolchain.rs

```diff
@@ -50,45 +51,10 @@ pub struct Toolchain {
     pub path: PathBuf,
 }
 
-#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
-pub struct TargetTriple(String);
-
-impl fmt::Display for TargetTriple {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        self.0.fmt(f)
-    }
-}
-
-impl TargetTriple {
-    pub fn new(name: &str) -> Self {
-        Self(name.to_string())
-    }
-
-    pub fn from_host() -> Result<Self> {
-        let architecture = match std::env::consts::ARCH {
-            "aarch64" | "x86_64" => std::env::consts::ARCH,
-            unsupported_arch => bail!("Unsupported architecture: {}", unsupported_arch),
-        };
-        let vendor = match std::env::consts::OS {
-            "macos" => "apple",
-            _ => "unknown",
-        };
-        let os = match std::env::consts::OS {
-            "macos" => "darwin",
-            "linux" => "linux-gnu",
-            unsupported_os => bail!("Unsupported os: {}", unsupported_os),
-        };
-
-        let target_triple = format!("{}-{}-{}", architecture, vendor, os);
-
-        Ok(Self(target_triple))
-    }
-}
-
 impl Toolchain {
     pub fn new(name: &str, target: Option<String>) -> Result<Self> {
```

Comment:
```suggestion
    pub fn new(name: &str, target: Option<TargetTriple>) -> Result<Self> {
```
Maybe we should take a `TargetTriple` as an argument here now that we have the type for it?

Reply:
I decided to remove the `target` arg instead after re-looking at its usage, because right now we are only passing `None` anyway. I was thinking to use an `Option` since in future we might want to support installation of cross-platform toolchains, but we can always extend this in future when that use-case comes around.